engines/dradis-api/app/controllers/dradis/ce/api/v1/attachments_controller.rb

```diff
@@ -0,0 +1,101 @@
+module Dradis::CE::API
+  module V1
+    class AttachmentsController < Dradis::CE::API::V1::ProjectScopedController
+      before_action :set_node
+
+      skip_before_action :json_required, :only => [:create]
+
+      def index
+        @attachments = @node.attachments.each(&:close)
+      end
+
+      def show
+        filename = params[:id]
+        begin
+          @attachment = Attachment.find(filename, conditions: { node_id: @node.id } )
+        rescue
+          raise ActiveRecord::RecordNotFound, "Couldn't find attachment with id '#{params[:id]}'"
+        end
+      end
+
+      def create
+        uploaded_files = params.fetch('files', [])
+
+        @attachments = []
+        uploaded_files.each do |uploaded_file|
+          attachment_name = get_name(original: uploaded_file.original_filename)
+
+          attachment = Attachment.new(attachment_name, node_id: @node.id)
+          attachment << uploaded_file.read
+          attachment.save
+
+          @attachments << attachment
+        end
+
+        if @attachments.any? && @attachments.count == uploaded_files.count
+          render status: 201
+        else
+          render status: 422
+        end
+      end
+
+      def update
+        filename    = params[:id]
+        attachment  = Attachment.find(filename, conditions: { node_id: @node.id } )
+        attachment.close
+
+        begin
+          new_name    = CGI::unescape(params[:attachment][:filename])
+          destination = Attachment.pwd.join(params[:node_id], new_name).to_s
+
+          if !File.exist?(destination) && !destination.match(/^#{Attachment.pwd}/).nil?
+            File.rename attachment.fullpath, destination
+            @attachment = Attachment.find(new_name, conditions: { node_id: @node.id } )
+          end
+        rescue
+          @attachment = attachment
+          render status: 422
+        end
+      end
+
+      def destroy
+        filename = params[:id]
+
+        @attachment = Attachment.find(filename, conditions: { node_id: @node.id} )
+        @attachment.delete
+        
+        render_successful_destroy_message
+      end
+
+      private
+
+      def set_node
+        @node = Node.find(params[:node_id])
+      end
+
+      def attachment_params
+        params.require(:files).permit()
+      end
+
+      # Obtain a suitable attachment name for the recently uploaded file. If the
+      # original file name is still available, use it, otherwise, provide count-based
+      # an alternative.
+      def get_name(args={})
```

Comment:
I feel that this should have a more meaningful name since, for the most part, you're not actually "getting" a name. Also, what's the reasoning behind using a hash argument?